.github/workflows/ci.yml

```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Print build information
-        run: "echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}"
+        run: 'echo head_ref: "$GITHUB_HEAD_REF", ref: "$GITHUB_REF"'
```

Comment:
You may have to set these as environment variables first

Reply:
I kind of can't believe that gha has such a security-footgun with no warnings or easy alternatives. I've fixed this in multiple other places. there isn't even a "shell-escape" filter available on substitutions :(

Reply:
Ah, sorry! I confused the exec line for the output. Thanks, will merge.